### PR TITLE
Remove more ActionType subclasses and dry up Acknowleded and Empty Response versions

### PR DESCRIPTION
**Pull request by original-brownbear**
It's in the title drying this stuff up some more.

### CODE REVIEW

**Comment by joegallo** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetDesiredBalanceAction.java [hunk 1]
```diff
@@ -59,7 +64,7 @@ public TransportGetDesiredBalanceAction(
         WriteLoadForecaster writeLoadForecaster
     ) {
         super(
-            GetDesiredBalanceAction.NAME,
+            "cluster:admin/desired_balance/get",
```
Comment (joegallo) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetDesiredBalanceAction.java:67]: `TYPE.name()`, yes?

**Comment by joegallo** [comment]
`RetentionLeaseActions` gets a bit jankier as result of this -- can you spin some cycles on re-beautifying it?

**Comment by joegallo** [comment]
Something for a different PR (not this one!):

```
    public static class Request extends AcknowledgedRequest<Request> {
        public Request() {}

        public Request(StreamInput in) throws IOException {
            super(in);
        }

        @Override
        public ActionRequestValidationException validate() {
            return null;
        }
    }
```

These classes seem to be appear a lot, and I'm not sure they're adding much value when they do appear -- is there anything that can be done about that?

File: server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetDesiredBalanceAction.java (hunk 1 above)
Comment (joegallo) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetDesiredBalanceAction.java]: 👍, resolved in 290531fcbcb34917d469af0d5aed039956640f9f

**Comment by joegallo** [comment]
Generally LGTM, once you revisit `RetentionLeaseActions` I think this is mergeable. Thanks for slogging through these!

**Comment by original-brownbear** [comment]
Thanks @joegallo see https://github.com/elastic/elasticsearch/pull/103242/commits/0c314c32ba57e4ee5f2746b3e02d798100dedf6f :)

**Review by joegallo — APPROVED** [review]
LGTM!

**Comment by original-brownbear** [comment]
Thanks Joe!

**Pull request merged**